build/gen_stub.php

```diff
@@ -1970,6 +1991,10 @@ public static function getClassSynopsisFilename(Name $name): string {
         return strtolower(implode('-', $name->parts));
     }
 
+    public static function getClassSynopsisReference(Name $name): string {
```

Comment:
I created a different method from `getClassSynopsisFilename()`, because maybe it's  slightly more clear this way that it's a reference, not a filename. Also, a few classes (mostly exceptions) sometimes use a different format (e.g. exception.synopsis), although I kept using the original `class.NAME` one, since apparently it always works.